packages/api-tests/test/window.core.spec.js

```diff
@@ -331,6 +358,74 @@ describe('WindowCore API', function(done) {
       return chainPromises(steps);
     });
 
+    it('Should add a listener to the window that persists after it has been called #ssf.Window.on #ssf.WindowCore.on', function() {
+      const windowTitle = 'windownameon';
+      const windowOptions = getWindowOptions({
+        name: windowTitle
+      });
+
+      const addWindowListener = (event) => {
+        /* eslint-disable no-undef */
+        const script = (event, callback) => {
+          var currentWin = ssf.Window.getCurrentWindow();
+          currentWin.on(event, () => console.log(event));
+          callback();
+        };
+        /* eslint-enable no-undef */
+        return executeAsyncJavascript(app.client, script, event);
+      };
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => addWindowListener('focus'),
+        () => callWindowMethod('listenerCount', 'focus'),
+        (result) => assert.equal(result.value, 1),
+        // Blur isn't working correctly in browser tests, so we need to switch windows instead
+        () => selectWindow(app.client, 0),
+        () => selectWindow(app.client, 1),
+        () => callAsyncWindowMethod('blur'),
+        () => callAsyncWindowMethod('focus'),
+        () => callWindowMethod('listenerCount', 'focus'),
+        (result) => assert.equal(result.value, 1)
+      ];
+
+      return chainPromises(steps);
+    });
+
+    it('Should add a one-time listener to the window #ssf.Window.once #ssf.WindowCore.once', function() {
+      const windowTitle = 'windownameonce';
+      const windowOptions = getWindowOptions({
+        name: windowTitle
+      });
+
+      const addWindowListener = (event) => {
+        /* eslint-disable no-undef */
+        const script = (event, callback) => {
+          var currentWin = ssf.Window.getCurrentWindow();
+          currentWin.once(event, () => console.log(event));
+          callback();
+        };
+        /* eslint-enable no-undef */
+        return executeAsyncJavascript(app.client, script, event);
+      };
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => addWindowListener('focus'),
+        () => callWindowMethod('listenerCount', 'focus'),
+        (result) => assert.equal(result.value, 1),
+        // Blur isn't working correctly in browser tests, so we need to switch windows instead
+        () => selectWindow(app.client, 0),
+        () => selectWindow(app.client, 1),
+        () => callAsyncWindowMethod('blur'),
+        () => callAsyncWindowMethod('focus'),
+        () => callWindowMethod('listenerCount', 'focus'),
+        (result) => assert.equal(result.value, 0)
+      ];
+
+      return chainPromises(steps);
+    });
+
```

Comment:
Is it worth including tests for removeListener and removeAllListeners (the latter with and without an event name)?